uaclient/config.py

```diff
@@ -577,9 +444,10 @@ def delete_cache_key(self, key: str) -> None:
             raise RuntimeError(
                 "Invalid or empty key provided to delete_cache_key"
             )
-        if key.startswith("machine-access") or key == "machine-token":
-            self._entitlements = None
-            self._machine_token = None
+        if key.startswith("machine-access"):
+            # will find a better way for this
+            self.machine_token_file._entitlements = None
+            self.machine_token_file._machine_token = None
```

Comment:
This can probably be just
```suggestion
            self._machine_token_file = None
```

Reply:
Oh yes, makes sense.